### ISSUE TITLE
Feature request: Termination/Deletion protection

### ISSUE THREAD

**Issue by jannylund** [rancher/rancher#16328]
The simplicity of Rancher means that people with little to no knowledge of Kubernetes are able to do things they could not before. However, with simplicity comes also a great amount of risks, since you can essentially delete a cluster in two clicks.

Suggestion:
- Possibility to enable deletion protection on:
  - cluster level
  - node level
  - project level
  - namespace level
  - deployment level
  - ingress level

Basically this could simply be on Rancher level so that the ui would not allow direct deletion without going to the process of disabling deletion protection.

**Comment by superseb** [comment]
Duplicate of https://github.com/rancher/rancher/issues/15075

**Comment by jannylund** [comment]
@superseb this is actually not a duplicate as it reflects features also on levels that are not part of the ticket refer to. Can you please reopen it to keep it logged?

**Comment by azbpa** [comment]
@superseb and @jannylund would also agree that it is not a duplicate.

Would like to see some kind of termination protection, at least at cluster level, to protect accidental deletion of it.

In my opinion there could be two strategies:
1. Introduce a flag, once it is checked, you cannot delete the cluster from UI unless you uncheck it before in the options of the cluster. 
2. Necessity to confirm a deletion by entering the name of the cluster that should be deleted.
